app/serializers/notice_serializer.rb

```diff
@@ -3,20 +3,40 @@ class NoticeSerializer < ActiveModel::Serializer
              :topics, :sender_name, :principal_name, :recipient_name, :works,
              :tags, :jurisdictions, :action_taken, :language
 
+  # Data to be returned when a Work has no associated URL.
+  FALLBACK = [{ url: 'No URL submitted' }].freeze
+
   # TODO: serialize additional entities
 
   def topics
     object.topics.map(&:name)
   end
 
+  # It would be cleaner to define a WorkSerializer and follow
+  # active-model-serializer conventions. However, we rename fields on Work in
+  # our API responses, conditionally upon the type of Notice associated. That
+  # is not a use case anticipated by active-model-serializer, and it turns out
+  # to be much easier to define the work attribute here, and then have it
+  # accessible as a hash (rather than as an object to be serialized) so we can
+  # use hash operations on it within the hooks supported by
+  # active-model-serializer.
   def works
-    object.works.as_json(
+    base_works = object.works.as_json(
       only: [:description],
       include: {
         infringing_urls: { only: %i[url url_original] },
         copyrighted_urls: { only: %i[url url_original] }
       }
     )
+    cleaned_works(base_works)
+  end
+
+  def cleaned_works(base_works)
+    base_works.each do |work|
+      work['infringing_urls'] = FALLBACK if work['infringing_urls'].empty?
+      work['copyrighted_urls'] = FALLBACK if work['copyrighted_urls'].empty?
```

Comment:
This doesn't work for some notice types.

![Screenshot from 2019-03-14 18-55-04](https://user-images.githubusercontent.com/2022788/54380118-06344c80-468b-11e9-8398-d7f91ac4cb7c.png)

Reply:
`base_works.each do {something}` does nothing when `base_works` is `[]`. Do you see the fallback text for a notice that has associated works (but when the works lack URLs)?

Reply:
Yeah, I was disconnecting works, not work urls. This looks fine.